tests/integration/test_charm.py

```diff
@@ -37,6 +38,35 @@ async def test_build_and_deploy(ops_test: OpsTest):
     await ops_test.model.wait_for_idle(timeout=60 * 10)
 
 
+@retry(
+    wait=wait_exponential(multiplier=1, min=1, max=10),
+    stop=stop_after_delay(60),
+    reraise=True,
+)
+async def connect_client_to_server_with_retry(
```

Comment:
Would it be possible to retry the existing `connect_client_to_server(...)` instead of creating a whole new method?

Reply:
I believe the equivalent alternative is:

```
retryer = Retrying(stop=stop_after_delay(60), ...)

def test_that_uses_connect_client_to_server(...):
    ...
    retryer(connect_client_to_server, "some error message if we fail")
```

Would that be preferable?

Reply:
Probably retrying the test is not the best idea since what we want to retry is the connection not the test itself.
It seems to me the abstraction of the `connect_client_to_server` method  from the retry is not needed as we are not using the connection w/o retrying anywhere else, and we are duplicating some code. I'm not saying it is wrong, it's probably just an embedded vs abstracted design.

I was thinking on something like this:

```python
async def connect_client_to_server(...):
    ...
    url = f"http://{service_name}.{model_name}.svc.cluster.local:{port}"
    @tenacity.retry(...)
    def connect_attempt(...):
        minio_cmd=...
        kubectl_cmd=....
    return ...
```